config/src/loaders/toml.rs

```diff
@@ -0,0 +1,106 @@
+//! Load the configuration from a file or a `String` written in [Toml format](Tomlhttps://en.wikipedia.org/wiki/TOML)
+
+use crate::Config;
+use std::fmt;
+use std::fs::File;
+use std::io;
+use std::io::Read;
+use std::result;
+use toml;
+
+/// Error type denoting the different errors this module can fail with.
+/// Parsing the configuration from Toml might fail with a
+/// `toml::de::Error`, but loading that configuration from a file
+/// might also fail with a `std::io::Error`.
+#[derive(Debug)]
+pub enum Error {
+    /// Indicates there was an error when trying to load a file from a file.
+    IOError(io::Error),
+    /// Indicates there was an error when trying to build a
+    /// `witnet_config::Config` instance out of the Toml string given.
+    ParseError(toml::de::Error),
+}
+
+/// Formats the error in a user-friendly manners. Suitable for telling
+/// the user what error happened when loading/parsing the
+/// configuration.
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::IOError(e) => e.fmt(f),
+            Error::ParseError(e) => e.fmt(f),
+        }
+    }
+}
+
+/// Just like `std::result::Result` but withe error param fixed to
+/// `Error` type in this module.
+pub type Result<T> = result::Result<T, Error>;
```

Comment:
Really like this error definition! However, I wonder if the error handling in [util](https://github.com/witnet/witnet-rust/blob/master/util/src/error.rs) should be used instead. I don't know if it is supposed that all the subcrates results should be `WitnetResult` like in the [storage library](https://github.com/witnet/witnet-rust/blob/master/storage/src/error.rs). Any thoughts @witnet/devs ?

Reply:
I don't see how I can use that currently, because in my case I'm not just defining a custom error, but wrapping two already existing ones: `std::io::Error` and `toml::de::Error`